shotover-proxy/src/transforms/mod.rs

```diff
@@ -521,6 +577,64 @@ pub trait Transform: Send {
     /// Wrapping it in an [`Arc<Mutex<_>>`](std::sync::Mutex) would make it a global count of all messages seen by this transform.
     async fn transform<'a>(&'a mut self, message_wrapper: Wrapper<'a>) -> ChainResponse;
 
+    /// This method should be should be implemented by your transform if it is required to process pushed messages (typically events
+    /// or messages that your source is subscribed to. The wrapper object contains the queries/frames
+    /// in a [`Vec<Message`](crate::message::Message).
+    ///
+    /// This transform method is not the same request/response model as the other transform method.
+    /// This method processes one pushed message before sending it in reverse on the chain back to the source.
+    ///
+    /// You can modify the messages in the wrapper struct to achieve your own designs. Your transform can
+    /// also modify the response from `message_wrapper.call_next_transform_pushed` if it needs to. As long as the message
+    /// carries on through the chain, it will function correctly. You are able to add or remove messages as this method is not expecting
+    /// request/response pairs.
+    ///
+    /// ## Invariants
+    /// * _Non-terminating_ - Your `transform_pushed` method should not be terminating as the messages should get passed back to the source, where they will terminate.
+    ///
+    ///
+    /// A basic reverse transform that logs event data and counts the number of events it sees could be defined like so:
+    /// ```
+    /// use shotover_proxy::transforms::{Transform, Wrapper};
+    /// use async_trait::async_trait;
+    /// use tracing::info;
+    /// use shotover_proxy::error::ChainResponse;
+    ///
+    /// #[derive(Debug, Clone)]
+    /// pub struct Printer {
+    ///     counter: i32,
+    /// }
+    ///
+    /// impl Default for Printer {
+    ///     fn default() -> Self {
+    ///         Self::new()
+    ///     }
+    /// }
+    ///
+    /// impl Printer {
+    ///     pub fn new() -> Printer {
+    ///         Printer { counter: 0 }
+    ///     }
+    /// }
+    ///
+    /// #[async_trait]
+    /// impl Transform for Printer {
+    ///     async fn transform<'a>(&'a mut self, message_wrapper: Wrapper<'a>) -> ChainResponse {
```

Comment:
This is documenting `transform` in the `transform_pushed` docs.

Maybe there should just be a single code snippet example on the trait itself?

Although maybe we dont need any of these code snippets at all, we could always just refer to an actual transform implementation? When we actually provide custom transforms to users I imagine the API will look quite different and we can create appropriate examples at that time.

Reply:
I agree with that. I'll remove it for now, no need to keep them as no one is reading the docs.rs at the moment.